### ISSUE TITLE
[css-grid-3][masonry] Clarify what mixing subgridded items with automatic and definite grid position does

### ISSUE THREAD

**Issue by ethanjv** [w3c/csswg-drafts#10926]
[The following section of the spec](https://drafts.csswg.org/css-grid-3/#track-sizing-subgrid) states how to handle automatically placed subgridded items within a masonry container, which are supposed to be consider in every possible track for sizing.

However, there is nothing clarifying the expectation for items with a definite grid position that are subgridded through an automatically placed submasonry. Is the expectation to have that definite position to be consider in "all tracks" within the parent grid axis, considering that it would be offset to the "possible" definite positions it could take?

If this is the expectation, then computing the "possible" tracks an item could be placed becomes very complex when multiple levels of subgridding occur, and it is relevant to have those resolved in order to create the virtual masonry items for [the optimized algorithm](https://drafts.csswg.org/css-grid-3/#track-sizing-performance).

**Comment by tabatkins** [comment]
Hm, I think our intent was just that definite-placed items affected the size of the tracks they could theoretically have gone into. So, like, say you have masonry with 5 tracks, and a span-3 submasonry, with an item placed explicitly in track 2 of the submasonry. The submasonry can go in any of the tracks, so 1-3, 2-4, or 3-5; the explicitly placed item, then, can be in the parent masonry's tracks 2, 3, or 4, and thus would be taken into account for those tracks' sizing.

I think that's what you're describing, right?

**Comment by fantasai** [comment]
We probably should consider this in conjunction with some of the auto-placement improvements in https://github.com/w3c/csswg-drafts/issues/10265.

Here's an example that I think we will want to make work:
* 6 tracks in pairs, type A and type B.
* auto-placed masonry parent grid spanning two tracks, but only allowed to be placed in the pattern A-B, i.e. spanning tracks 1-2, 3-4, 5-6 and not 2-3 or 4-5.

I don't think we have very strong use cases for the OP case to work in a particular way, but for this case we want items that can only end up in A tracks to not contribute to B tracks and vice versa.

**Comment by alisonmaher** [comment]
> I think that's what you're describing, right?

@tabatkins Yeah, I believe the question is around what to do for explicit placed items, since the spec only says what to do for auto placed items. Your suggestion makes sense, and I believe that would allow for extension to the case that @fantasai brings up in https://github.com/w3c/csswg-drafts/issues/10926#issuecomment-3348988361

**Comment by alisonmaher** [comment]
Per this resolution, I wanted to confirm the behavior of a test I added with multiple nested layers of auto placed subgrids. I'll use [column-subgrid-margin-006.html](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/web_tests/external/wpt/css/css-grid/grid-lanes/subgrid/grid-subgridded-to-grid-lanes/track-sizing/column-subgrid-extra-margin-006.html?q=third_party%2Fblink%2Fweb_tests%2Fexternal%2Fwpt%2Fcss%2Fcss-grid%2Fgrid-lanes%2Fsubgrid%2Fgrid-subgridded-to-grid-lanes%2Ftrack-sizing%2Fcolumn-subgrid-extra-margin-006.html&ss=chromium) as an example (but you can find all relevant tests related to this question in [Chromium CL:7853073](https://chromium-review.googlesource.com/c/chromium/src/+/7853073))

This test has three layers of auto placed subgrids inside a grid lanes container. In a grid container, we know the subgrid placements ahead of time, so the padding of the leaves only contributes the tracks they are placed in, resulting in a 100x100px green square (the math happens to be that track 1 gets 29px, track 2 is 23px, track 3 is 27px, and track 4 is 21px).

In grid lanes, though, in Chromium, we treat each of these layers as "auto placed", which means we can never get the same 100px as grid because we end up placing the largest contribution in every possible track. What we end up getting in Chromium is 29px coverage in all 4 tracks, leading to a 116px wide grid lanes container.

Granted, the 29px contribution technically isn't possible at every position, because the outermost subgrid can only be placed in one position, but to avoid incurring the cost of placing every item in every possible position in a nested context, and taking the max from that, I ended up implementing this to place the largest contribution in every position, similar to the virtual item optimization.

Does this interpretation make sense? Is this the correct expectation for this test?

CC @tabatkins @fantasai

**Comment by tabatkins** [comment]
@alisonmaher fantasai and I reviewed the test, and we agree with your conclusions. We suggest, however, for the future, not to try to cram tests into green rectangles and instead focus on making them easy to understand. ^_^ The green square pattern is left over from when the tests had to be run manually, and needed to be fast to evaluate pass/fail visually.
